### ISSUE TITLE
#2987: Add new columns dialog: filter out irrelevant functions from suggestions dropdown

### ISSUE THREAD

**Issue by katerina-stepanova** [datagrok-ai/public#2987]
1.20.1, power pack 1.1.10

Function suggestions still contain too much irrelevant entries, that might be confusing for a regular user.
For example, packages:
<img width="250" alt="image" src="https://github.com/user-attachments/assets/dc3fbc6b-6409-4714-ad68-d9202112166a">
<img width="252" alt="image" src="https://github.com/user-attachments/assets/b6059e30-c20e-42fb-93e8-3d87083c4169">

**Expected results:**
Define which functions are actually useful in calculated columns, filter out all irrelevant ones.
For packages: are there any real use cases? If not, it might be more useful to filter it out completely.

**Comment by dnillovna** [comment]
This issue has been mirrored in Jira: https://reddata.atlassian.net/browse/GROK-16492

**Comment by LesiaPavlenko** [comment]
v. 1.21.1
PowerPack v.1.2.0
We are waiting for feedback about filtering

**Comment by katerina-stepanova** [comment]
v. 1.21.1
PowerPack v.1.2.0

There are still a lot of irrelevant functions:
- JS, DBScalarQuery, Call, DeployPackageVersion etc.
- functions from packages, e.g,.
<img width="1255" alt="image" src="https://github.com/user-attachments/assets/300788c1-23ab-466f-9dfe-cac5aae694f4">


Some of these appear in suggestions dropdown, some are present only in the right side panel.

**Comment by MariaDolotova** [comment]
Hi @katerina-stepanova , I agree regarding the follwoing functions: Call, DeployPackageVersion, JS. They should be marked as internal and not be included into the list. 

But as for DBScalarQuery - as far as I know, it can be used in the 'Add new column' dialog. As well as package functions. I mean, potentially any function which returns some scalar value (or a vector function which returns column and is marked by special tag) can be used in 'Add new column'.
Moreover, ability to use package functions straight from 'Add new column' dialog is an improvement itself.

Basically, we introduced different kinds of filtrations, sortings and hints to help user to navigate through the list of functions easily. Like autocompletion filtering as you type, functions filtering by categories (using checkboxes), functions search. Also if you click on a column in a column list, functions in the right widget will be sorted with the most relevant once on top.

Maybe I do not fully understand why would wider list of functions confuse users. Considering all available hints...
